x-pack/plugins/cloud_security_posture/public/components/setup_status.tsx

```diff
@@ -0,0 +1,187 @@
+    <FormattedMessage id="xpack.csp.setupStatus.loadingDescription" defaultMessage="Loading..." />
+  </CspLoadingState>
+);
+
+const defaultErrorRenderer = (error: unknown) => (
+  <EuiEmptyPrompt
+    css={css`
+      margin-top: 50px;
+    `}
+    color="danger"
+    iconType="alert"
+    data-test-subj={ERROR_STATE_TEST_SUBJECT}
+    title={
+      <h2>
+        <FormattedMessage
+          id="xpack.csp.setupStatus.errorRenderer.errorTitle"
+          defaultMessage="We couldn't fetch your cloud security posture data"
+        />
+      </h2>
+    }
+    body={
+      isCommonError(error) ? (
+        <p>
+          <FormattedMessage
+            id="xpack.csp.setupStatus.errorRenderer.errorDescription"
+            defaultMessage="{error} {statusCode}: {body}"
+            values={{
+              error: error.body.error,
+              statusCode: error.body.statusCode,
+              body: error.body.message,
+            }}
+          />
+        </p>
+      ) : undefined
+    }
+  />
+);
+
+const defaultNoDataRenderer = () => {
+  return (
+    <NoDataPage
+      data-test-subj={DEFAULT_NO_DATA_TEST_SUBJECT}
+      css={css`
+        margin-top: 50px;
+      `}
+      pageTitle={i18n.translate('xpack.csp.setupStatus.defaultNoDataConfig.pageTitle', {
+        defaultMessage: 'No data found',
+      })}
+      solution={i18n.translate('xpack.csp.setupStatus.defaultNoDataConfig.solutionNameLabel', {
+        defaultMessage: 'Cloud Security Posture',
+      })}
+      // TODO: Add real docs link once we have it
+      docsLink={'https://www.elastic.co/guide/index.html'}
+      logo={'logoSecurity'}
+      actions={{}}
+    />
+  );
+};
+
+interface SetupStatusProps<TData, TError> {
+  children: React.ReactNode;
+  query?: UseQueryResult<TData, TError>;
+  loadingRender?: () => React.ReactNode;
+  errorRender?: (error: TError) => React.ReactNode;
+  noDataRenderer?: () => React.ReactNode;
+}
+
+export const SetupStatus = <TData, TError>({
+  children,
+  query,
+  loadingRender = defaultLoadingRenderer,
+  errorRender = defaultErrorRenderer,
+  noDataRenderer = defaultNoDataRenderer,
+}: SetupStatusProps<TData, TError>) => {
+  const cisKubernetesPackageInfo = useCisKubernetesIntegration();
+  const cisIntegrationLink = useCISIntegrationLink();
+
+  const render = () => {
+    if (cisKubernetesPackageInfo.isError) {
+      return defaultErrorRenderer(cisKubernetesPackageInfo.error);
+    }
+
+    if (cisKubernetesPackageInfo.isLoading || cisKubernetesPackageInfo.isIdle) {
+      return defaultLoadingRenderer();
+    }
+
+    if (cisKubernetesPackageInfo.data.item.status !== 'installed') {
+      return packageNotInstalledRenderer(cisIntegrationLink);
+    }
+
+    if (!query) {
+      return children;
+    }
+
+    if (query.isError) {
+      return errorRender(query.error);
+    }
+
+    if (query.isLoading || query.isIdle) {
+      return loadingRender();
+    }
+
+    if (!query.data) {
+      return noDataRenderer();
+    }
```

Comment:
note that in the previous implementation, you could display an `EuiEmptyState` without neccessarly having to have no data from your query. I'll need to check that but if I'm not mistaken under those changes the `noDataRenderer` in the dashboard will not be displayed.

Reply:
You can do this here as well, but you need to do it in the child component and not in `SetupStatus`. You are correct, there is a bug here with the dashboard, I'll fix it